Neos.Flow/Classes/ResourceManagement/ResourceTypeConverter.php

```diff
@@ -306,8 +308,35 @@ protected function handleUploadedFile(UploadedFileInterface $source, PropertyMap
             return $this->convertedResources[spl_object_hash($source)];
         }
 
+        $stream = $source->getStream()->detach();
+
+        /**
+         * Validate file content and throw exception if content are suspicious
+         * For now this only checks svg file to the issue here https://nvd.nist.gov/vuln/detail/CVE-2023-37611
+         * @todo create a feature from this and allow to register code for checking uploaded files before importing
+         */
+        $pathInfo = UnicodeFunctions::pathinfo($source->getClientFilename());
+        if ($pathInfo['extension'] == 'svg') {
+            // @todo: Simplify again when https://github.com/darylldoyle/svg-sanitizer/pull/90 is merged and released.
+            $previousXmlErrorHandling = libxml_use_internal_errors(true);
+            $sanitizer = new Sanitizer();
+            $sanitizer->sanitize(stream_get_contents($stream));
+            libxml_clear_errors();
+            libxml_use_internal_errors($previousXmlErrorHandling);
+            $issues = $sanitizer->getXmlIssues();
+            if ($issues && count($issues) > 0) {
+                $messages = array_map(function (array $issue) {
+                    return $issue['message'] . ' in line ' . $issue['line'] . '.';
+                }, $issues);
+                $this->logger->warning('Blocked upload of suspicious file', ['issues' => $issues] + LogEnvironment::fromMethodName(__METHOD__));
+
+                return new FlowError('The upload of file "' . strip_tags($pathInfo['basename']) . '" was blocked. ' . implode(' ', $messages) . '  As it could pose a security risk it cannot be imported.', 1695361773);
+            }
+        }
```

Comment:
Looks good for me, but I would suggest to spend a method like `checkForVulnerableCodeInSVG()` or similar for better reading.